### PR TITLE
C_GetAttributeValue() - fix segfault

### PR DESCRIPTION
**Pull request by vjardin**
Revert the previous commit and update the PHPT in order to support
both a smartcard and the SoftHSM2.

### CODE REVIEW

**Comment by vjardin** [comment]
This pull request is a **placeholder** for a deeper analysis: I feel that it could be a good practice of not overwriting the template of C_GetAttributeValue() that's why the data is pushed/set into a 'Object' entry.

_Another option_: should we php/clone the Template so it can be updated using the Oasis' semantic ? Working on a clone seems to be the only alternative. If we do not clone it, because the Template is modified/updated under the hood, then the workflow of the PHP code becomes tricky because the users may feel that the Template shall remain unmodified when it is used within a loop (foreach return'd objects from C_FindObjects).

If we continue with this "clone option", currently, I do not know how to implement it while we maintain "simple" code that should remain easy to support.

**Comment by gamringer** [comment]
I disagree with the "object" entry solution. The point of this method is to replicate the OASIS API, so let's do just that.

I do agree that it's bad for the function to be working this way, but that one is on OASIS ;) If we want to not override the original template array, P11Object::getAttributeValue(array $attributeIds) is available, which returns a new array with the values.

Using a cloning method to keep the original template could be done in PHP land and does not require the function as we currently have it to change.

**Comment by vjardin** [comment]
@gamringer I am back to your array format along with the OASIS' semantic. This drop avoids the segfault.

File: tests/0290-oasis_FindObjects.phpt [hunk 1]
```diff
@@ -138,17 +135,7 @@ int(0)
...
 int(0)
-int(1)
-dump object %d: array(3) {
-  [16]=>
-  string(8) "PHP Test"
-  [17]=>
-  string(12) "Hello World!"
-  [3]=>
-  string(37) "Test Label - %A"
-}
```
Comment (gamringer) [inline tests/0290-oasis_FindObjects.phpt:149]: I think this test should definitely check for the object's attributes to ensure it's the correct object.

Comment (vjardin) [reply tests/0290-oasis_FindObjects.phpt:149]: The problem is that different HSMs lead to different output. I performed the test 0290 on another HSM that is not SoftHSM2, for this case, I cannot not get this output.

Same for the 0291, SoftHSM2 does not provide the same output.

I agree that it is a "weak" check, but it seems to be "good enough" : it was already helpful to detect some segfault. I prefer a "weak" test that can be improved later on than "no test".

Before having a "hardened" design, I'd like to use these tests in order to perform some more robustness analysis with Valgrind once they'll be set.

**Comment by vjardin** [comment]
Hi @gamringer ,

I see that you are working on a nice list of new tests. Before jumping on them, please, could you apply this serie ? Then, it'll be easier to rebase with your serie + I'll be able to check your serie with my PKCS11 environment.

Moreover, it'll help me to work on the hardened extension of the current tests.

**Comment by vjardin** [comment]
Thanks

**Pull request merged**